src/walletSdk/Types/recovery.ts

```diff
@@ -0,0 +1,157 @@
+import { Transaction } from "stellar-sdk";
+
+import { WalletSigner } from "walletSdk/Auth";
+import { AccountKeypair, PublicKeypair } from "walletSdk/Horizon";
+import { AuthToken } from "./auth";
+import { CommonBuilder } from "./horizon";
+
+/**
+ * Configuration for recoverable wallet
+ *
+ * @param accountAddress Stellar address of the account that is registering
+ * @param deviceAddress Stellar address of the device that is added as a primary signer. It will
+ * replace the master key of [accountAddress]
+ * @param accountThreshold Low, medium, and high thresholds to set on the account
+ * @param accountIdentity A list of account identities to be registered with the recovery servers
+ * @param signerWeight Signer weight of the device and recovery keys to set
+ * @param sponsorAddress optional Stellar address of the account sponsoring this transaction
+ */
+export type RecoverableWalletConfig = {
+  accountAddress: AccountKeypair;
+  deviceAddress: AccountKeypair;
+  accountThreshold: AccountThreshold;
+  accountIdentity: RecoveryIdentityMap;
+  signerWeight: SignerWeight;
+  sponsorAddress?: AccountKeypair;
+  builderExtra?: (builder: CommonBuilder) => CommonBuilder;
+};
+
+export type RecoverableWallet = {
+  transaction: Transaction;
+  signers: string[];
+};
+
+export type AccountSigner = {
+  address: AccountKeypair;
+  weight: number;
+};
+
+/**
+ * Account weights threshold
+ *
+ * @param low Low threshold weight
+ * @param medium Medium threshold weight
+ * @param high High threshold weight
+ */
+export type AccountThreshold = {
+  low: number;
+  medium: number;
+  high: number;
+};
+
+export type SignerWeight = {
+  device: number;
+  recoveryServer: number;
+};
+
+/**
+ * Recovery server configuration
+ *
+ * @property endpoint main endpoint (root domain) of SEP-30 recovery server. E.g.
+ * `https://testanchor.stellar.org`
+ * @property authEndpoint SEP-10 auth endpoint to be used. Should be in format
+ * `<https://domain/auth>`. E.g. `https://testanchor.stellar.org/auth`)
+ * @property homeDomain is a SEP-10 home domain. E.g. `testanchor.stellar.org`
+ * @property walletSigner optional [WalletSigner] used to sign authentication
+ */
+export type RecoveryServer = {
+  endpoint: string;
+  authEndpoint: string;
+  homeDomain: string;
+  walletSigner?: WalletSigner;
+  clientDomain?: string;
+};
+
+export type RecoveryServerKey = string;
```

Comment:
just curious can the key here be anything?

Reply:
this is a great question actually, I had stellar address in my head as the key but it appears this could be anything, like any kind of id string